src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -208,45 +307,91 @@ trait LazyScope
 {
   this: LazyModule =>
   override def toString: String = s"LazyScope named $name"
+  /** Preserve the previous value of the [[LazyModule.scope]], because when calling [[apply]] function,
+    * [[LazyModule.scope]] will be altered.
+    * */
   def apply[T](body: => T) = {
     val saved = LazyModule.scope
+    /** [[LazyModule.scope]] stack push. */
     LazyModule.scope = Some(this)
+    /** Evaluate [[body]] is this scope, save the return to [[out]]. */
     val out = body
+    /** Check the return scope is a right state. */
     require (LazyModule.scope.isDefined, s"LazyScope ${name} tried to exit, but scope was empty!")
     require (LazyModule.scope.get eq this, s"LazyScope ${name} exited before LazyModule ${LazyModule.scope.get.name} was closed")
+    /** [[LazyModule.scope]] stack pop. */
     LazyModule.scope = saved
     out
   }
 }
 
+/** Used to automatically create a level of module hierarchy (a [[SimpleLazyModule]]) within
+  * which [[LazyModule]]s can be instantiated and connected.
+  * It will instantiate a [[SimpleLazyModule]] to manage append `body`.
+  * And make `body` codes evaluated in this scope.
+  * */
 object LazyScope
 {
+  /** Create a [[LazyScope]] with name in its val name.
+    *
+    * @param body code need to be evaluated.
+    * */
```

Comment:
what's the scaladoc style for documenting `implicits`? In this case you mention the `val name` above but it doesn't actually show up in the scaladoc, and doesn't actually match any of the parameter names, which could either be confusing or convention... @seldridge , @jackkoenig thoughts on proper scaladoc here?